Release 3.5 constructs an incorrect job from JCL input file.

<!--
  Before opening a new issue, please search our existing issues: https://github.com/zowe/zowe-explorer-vscode/issues
-->

**Describe the bug**

<!-- A clear and concise description of what the bug is. -->

When submitting a JCL job that worked with prior version, job constructs incorrect job and fails when trying to submit it.  Rolled back Zowe Explorer to prior version and got expected behavior.

After submitting job, received new Popup box:   "Are you sure you want to submit the following job".  PTYJCSM
- Submit.

This is incorrect.  The job that I want to be submitted is 'JGATES.STAGE911.CNTL(PTYJCSM)' .

Log as below:

Zowe Explorer 3.5.0
This log file can be found at /Users/john.gates/.vscode/extensions/zowe.vscode-extension-for-zowe-3.5.0
Zowe Explorer log level: INFO
[2026/05/28 20:47:13] [INFO] Zowe Explorer profiles are being set as secured.
[2026/05/28 20:47:13] [INFO] Zowe home directory is located at /Users/john.gates/.zowe
[2026/05/28 20:47:13] [INFO] No custom credential managers found, using the default instead.
[2026/05/28 20:47:13] [INFO] Zowe Explorer profiles are being set as secured.
[2026/05/28 20:47:13] [INFO] Zowe Explorer is using the team configuration file "zowe.config.json"
[2026/05/28 20:47:13] [INFO] Zowe profiles initialized successfully.
[2026/05/28 20:47:13] [INFO] z/OSMF throttling set to 4 concurrent requests.
[2026/05/28 20:47:13] [INFO] z/OSMF queue timeout set to 300000 milliseconds.
[2026/05/28 20:47:13] [INFO] No custom credential managers found, using the default instead.
[2026/05/28 20:47:13] [INFO] Zowe Explorer profiles are being set as secured.
[2026/05/28 20:50:24] [INFO] Zowe Explorer has activated successfully.
[2026/05/28 20:52:28] [ERROR] Error: Rest API failure with HTTP(S) status 400
Data set not found: PTYJCSM
{
  errorDetails: [o [Error]: Rest API failure with HTTP(S) status 400
  Data set not found: PTYJCSM
  	at y.populateError (/Users/john.gates/.vscode/extensions/zowe.vscode-extension-for-zowe-3.5.0/out/src/vendors.extension.js:429:272)
  	at R (/Users/john.gates/.vscode/extensions/zowe.vscode-extension-for-zowe-3.5.0/out/src/vendors.extension.js:414:36282)
  	at y.onEnd (/Users/john.gates/.vscode/extensions/zowe.vscode-extension-for-zowe-3.5.0/out/src/vendors.extension.js:414:36604)
  	at IncomingMessage.<anonymous> (/Users/john.gates/.vscode/extensions/zowe.vscode-extension-for-zowe-3.5.0/out/src/vendors.extension.js:414:34642)
  	at IncomingMessage.emit (node:events:531:35)
  	at endReadableNT (node:internal/streams/readable:1698:12)
  	at process.processTicksAndRejections (node:internal/process/task_queues:89:21)] {
    mDetails: {
      msg: 'Rest API failure with HTTP(S) status 400\nData set not found: PTYJCSM',
      causeErrors: '{"rc":4,"reason":16,"category":6,"message":"Data set not found: PTYJCSM"}',
      source: 'http',
      errorCode: 400,
      protocol: 'https',
      port: 10443,
      host: '<host>',
      basePath: '',
      httpStatus: 400,
      errno: undefined,
      syscall: undefined,
      payload: { file: "//'PTYJCSM'" },
      headers: [
        { 'Content-Type': 'application/json' },
        { 'X-CSRF-ZOSMF-HEADER': true }
      ],
      resource: '/zosmf/restjobs/jobs',
      request: 'PUT',
      additionalDetails: 'Received HTTP(S) error 400 = Bad Request.\n' +
        '\n' +
        'Protocol:          https\n' +
        'Host:              <host>\n' +
        'Port:              <port>\n' +
        'Base Path:         \n' +
        'Resource:          /zosmf/restjobs/jobs\n' +
        'Request:           PUT\n' +
        'Headers:           [{"Content-Type":"application/json"},{"X-CSRF-ZOSMF-HEADER":true}]\n' +
        `Payload:           { file: "//'PTYJCSM'" }\n` +
        'Allow Unauth Cert: true\n' +
        'Available creds:   user,password\n' +
        'Your auth order:   basic,token,bearer,cert-pem\n' +
        'Auth type used:    basic'
    },
    mMessage: 'Rest API failure with HTTP(S) status 400\nData set not found: PTYJCSM'
  },
  apiType: 'mvs',
  profile: undefined,
  scenario: 'Job submission failed.'
}


**To Reproduce**

<!--
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
-->

**Expected behavior**

<!-- A clear and concise description of what you expected to happen. -->
I expect the correct job to be submitted.

**Screenshots**

<!-- If applicable, add screenshots to help explain your problem. -->

**Desktop (please complete the following information):**

- OS:  MAC/OS 26.5
- Zowe Explorer Version:  3.5
- (Optional) Zowe CLI Version:
- (Optional) Do you have secure credentials enabled?
- (Optional) Are you using a custom credential manager?

**Additional context**

<!-- Add any other context about the problem here. -->

--- a/packages/zowe-explorer/CHANGELOG.md
+++ b/packages/zowe-explorer/CHANGELOG.md
@@ -4,6 +4,12 @@ All notable changes to the "vscode-extension-for-zowe" extension will be documen
 
 ## TBD Release
 
+### Bug fixes
+
+- Fixed an issue where submitting a job from a favorited PDS member would fail. [#4286](https://github.com/zowe/zowe-explorer-vscode/pull/4286)
+
+## `3.5.0`
+
 ### New features and enhancements
 
 - Enhanced "Show Attributes" feature with improved visual presentation: aligned all attribute table columns consistently across core and extender sections, added thousands separators for numeric values, and automatically appends the `%` symbol to "Used Space" values for better readability. [#3927](https://github.com/zowe/zowe-explorer-vscode/issues/3927)

--- a/packages/zowe-explorer/CHANGELOG.md
+++ b/packages/zowe-explorer/CHANGELOG.md
@@ -6,7 +6,7 @@ All notable changes to the "vscode-extension-for-zowe" extension will be documen
 
 ### Bug fixes
 
-- Fixed an issue where submitting a job from a favorited PDS member would fail. [#4286](https://github.com/zowe/zowe-explorer-vscode/pull/4286)
+- Fixed an issue where submitting a job from a favorited PDS member would fail. [#4282](https://github.com/zowe/zowe-explorer-vscode/issues/4282)
 
 ## `3.5.0`
 

--- a/packages/zowe-explorer/__tests__/__unit__/trees/dataset/DatasetActions.unit.test.ts
+++ b/packages/zowe-explorer/__tests__/__unit__/trees/dataset/DatasetActions.unit.test.ts
@@ -435,6 +435,22 @@ describe("Dataset Actions Unit Tests - Function refreshPS", () => {
         await DatasetActions.refreshPS(child);
         expect(blockMocks.fetchDsAtUri).toHaveBeenCalledWith(child.resourceUri, { editor: undefined });
     });
+    it("Checking favorite PDS Member refresh when member node context has favorite suffix", async () => {
+        createGlobalMocks();
+        const blockMocks = createBlockMocksShared();
+        const parent = new ZoweDatasetNode({
+            label: "parent",
+            collapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
+            parentNode: blockMocks.datasetSessionNode,
+        });
+        const child = new ZoweDatasetNode({ label: "child", collapsibleState: vscode.TreeItemCollapsibleState.None, parentNode: parent });
+        parent.contextValue = Constants.DS_PDS_CONTEXT + Constants.FAV_SUFFIX;
+        child.contextValue = Constants.DS_MEMBER_CONTEXT + Constants.FAV_SUFFIX;
+
+        await DatasetActions.refreshPS(child);
+        
+        expect(blockMocks.fetchDsAtUri).toHaveBeenCalledWith(child.resourceUri, { editor: undefined });
+    });
     it("Checking favorite PS refresh", async () => {
         createGlobalMocks();
         const blockMocks = createBlockMocksShared();

--- a/packages/zowe-explorer/__tests__/__unit__/trees/job/JobActions.unit.test.ts
+++ b/packages/zowe-explorer/__tests__/__unit__/trees/job/JobActions.unit.test.ts
@@ -968,6 +968,42 @@ describe("Jobs Actions Unit Tests - Function submitMember", () => {
             "Job submitted: [TESTJOB(JOB1234)](command:zowe.jobs.setJobSpool?%5B%22test%22%2C%22JOB1234%22%5D)"
         );
     });
+    it("Checking Submit Job for PDS Member with favorite suffix context", async () => {
+        createGlobalMocks();
+        const blockMocks = createBlockMocks();
+        mocked(Profiles.getInstance).mockReturnValue(blockMocks.profileInstance);
+        const favProfileNode = new ZoweDatasetNode({
+            label: "test",
+            collapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
+            parentNode: blockMocks.datasetSessionNode,
+        });
+        favProfileNode.contextValue = Constants.FAV_PROFILE_CONTEXT;
+        const favoriteSubNode = new ZoweDatasetNode({
+            label: "TEST.JCL",
+            collapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
+            parentNode: favProfileNode,
+        });
+        favoriteSubNode.contextValue = Constants.DS_PDS_CONTEXT + Constants.FAV_SUFFIX;
+        const favoriteMember = new ZoweDatasetNode({
+            label: "MEMBER1",
+            collapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
+            parentNode: favoriteSubNode,
+        });
+        // Set the context value with the favorite suffix to trigger the bug condition we fixed
+        favoriteMember.contextValue = Constants.DS_MEMBER_CONTEXT + Constants.FAV_SUFFIX;
+        const submitJobSpy = vi.spyOn(blockMocks.jesApi, "submitJob");
+        submitJobSpy.mockClear();
+        submitJobSpy.mockResolvedValueOnce(blockMocks.iJob);
+
+        await DatasetActions.submitMember(favoriteMember);
+        expect(submitJobSpy).toHaveBeenCalled();
+        // Since the member is in a PDS, even if the member itself is favorited, the label should still be PDS(member)
+        expect(submitJobSpy.mock.calls[0][0]).toEqual("TEST.JCL(MEMBER1)");
+        expect(mocked(Gui.showMessage)).toHaveBeenCalled();
+        expect(mocked(Gui.showMessage).mock.calls[0][0]).toEqual(
+            "Job submitted: [TESTJOB(JOB1234)](command:zowe.jobs.setJobSpool?%5B%22test%22%2C%22JOB1234%22%5D)"
+        );
+    });
     it("Checking Submit Job for Favourite PS Dataset content", async () => {
         createGlobalMocks();
         const blockMocks = createBlockMocks();

--- a/packages/zowe-explorer/__tests__/__unit__/trees/job/JobActions.unit.test.ts
+++ b/packages/zowe-explorer/__tests__/__unit__/trees/job/JobActions.unit.test.ts
@@ -989,7 +989,7 @@ describe("Jobs Actions Unit Tests - Function submitMember", () => {
             collapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
             parentNode: favoriteSubNode,
         });
-        // Set the context value with the favorite suffix to trigger the bug condition we fixed
+
         favoriteMember.contextValue = Constants.DS_MEMBER_CONTEXT + Constants.FAV_SUFFIX;
         const submitJobSpy = vi.spyOn(blockMocks.jesApi, "submitJob");
         submitJobSpy.mockClear();
@@ -997,7 +997,7 @@ describe("Jobs Actions Unit Tests - Function submitMember", () => {
 
         await DatasetActions.submitMember(favoriteMember);
         expect(submitJobSpy).toHaveBeenCalled();
-        // Since the member is in a PDS, even if the member itself is favorited, the label should still be PDS(member)
+
         expect(submitJobSpy.mock.calls[0][0]).toEqual("TEST.JCL(MEMBER1)");
         expect(mocked(Gui.showMessage)).toHaveBeenCalled();
         expect(mocked(Gui.showMessage).mock.calls[0][0]).toEqual(

--- a/packages/zowe-explorer/src/trees/dataset/DatasetActions.ts
+++ b/packages/zowe-explorer/src/trees/dataset/DatasetActions.ts
@@ -2000,7 +2000,7 @@ export class DatasetActions {
             const defaultMessage = vscode.l10n.t("Cannot submit, item invalid.");
             switch (true) {
                 // For favorited or non-favorited sequential DS:
-                case SharedContext.isFavorite(node):
+                case SharedContext.isFavorite(node) && !SharedContext.isDsMember(node):
                 case SharedContext.isSessionNotFav(node.getParent()):
                     sesName = node.getParent().getLabel() as string;
                     label = node.label as string;
@@ -2188,7 +2188,7 @@ export class DatasetActions {
         try {
             switch (true) {
                 // For favorited or non-favorited sequential DS:
-                case SharedContext.isFavorite(node):
+                case SharedContext.isFavorite(node) && !SharedContext.isDsMember(node):
                 case SharedContext.isSessionNotFav(node.getParent()):
                 case SharedContext.isDs(node):
                     label = node.label as string;

